opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java

```diff
@@ -70,7 +78,34 @@ public PaymentInfoBody extractPaymentInfoBody(ProcessResponse result) {
                 target = "resourceId"
         )
         @Mapping(source = "accountNumber", target = "bban")
+        @Mapping(source = "balances", target = "balances", qualifiedByName = "hbciBalancesToProtocolBalances")
         AccountListDetailBody map(BankAccount account);
+
+
+        @Named("hbciBalancesToProtocolBalances")
+        static List<Balance> hbciBalancesToProtocolBalances(BalancesReport hbciBalances) {
+            List<Balance> protocolBalances = new ArrayList<>();
+            if (hbciBalances != null) {
```

Comment:
You can immediately return protocolBalances here

```
if (hbciBalances == null) {
	return protocolBalances;
}

add(protocolBalances, "available", hbciBalances.getAvailableBalance());
add(protocolBalances, "credit", hbciBalances.getCreditBalance());
add(protocolBalances, "ready", hbciBalances.getReadyBalance());
add(protocolBalances, "unready", hbciBalances.getUnreadyBalance());
add(protocolBalances, "used", hbciBalances.getUsedBalance());

return protocolBalances;
```

Reply:
done